runtime-transaction/src/svm_transaction_adapter.rs

```diff
@@ -0,0 +1,25 @@
+use {
+    core::borrow::Borrow,
+    solana_sdk::transaction::{SanitizedTransaction, VersionedTransaction},
+    solana_svm_transaction::svm_transaction::SVMTransaction,
+};
+
+/// Adapter trait to allow for conversion to legacy transaction types
+/// that are required by some outside interfaces.
+/// Eventually this trait should go away, but for now it is necessary.
+pub trait SVMTransactionAdapter: SVMTransaction {
+    fn as_sanitized_transaction(&self) -> impl Borrow<SanitizedTransaction>;
```

Comment:
I think it makes more sense to have a `to_sanitized_transaction` method which does the clone internally since we only really need this adapter method for clones. The `commit_transactions` call-site for geyser can use `deref` directly

Reply:
`commit_transactions` is where we actually need/want this as borrow.

Because we need to have a `SanitizedTransaction` and if we don't we need to convert to one - it can't just be deref once we are generic.

Reply:
but doing the borrow there also let's us **not** clone the the SanitizedTransaction if that's what we already have